src/Inventory/Conf.php

```diff
@@ -935,38 +935,54 @@ public function showConfigForm()
             );
             //if action == action_status => show blocation else hide blocaction
             echo Html::scriptBlock("
-            function changestatus() {
-                if ($('#dropdown_stale_agents_action$rand').val() != 0) {
-                $('#blocaction1').show();
-                $('#blocaction2').show();
-                } else {
-                $('#blocaction1').hide();
-                $('#blocaction2').hide();
+                function changestatus() {
+                    if ($('#dropdown_stale_agents_action$rand').val() != 0 && $('#dropdown_stale_agents_action$rand').val() != 2) {
+                        $('#bloc_status_action1').show();
+                        $('#bloc_status_action2').show();
+                    } else {
+                        $('#bloc_status_action1').hide();
+                        $('#bloc_status_action2').hide();
+                    }
                 }
-            }
-            changestatus();
-
-        ");
+                changestatus();
+            ");
             echo "</td>";
             echo "</tr>";
             //blocaction with status
-            echo "<tr class='tab_bg_1'><td colspan=2></td>";
+            echo "<tr class='tab_bg_1' style='display:none' id='bloc_status_action1'><td colspan=2></td>";
             echo "<td>";
-            echo "<span id='blocaction1' style='display:none'>";
-            echo \State::createTabEntry(__('Change the status'), 0, \State::getType());
-            echo "</span>";
+            echo \State::createTabEntry(__('If the asset status is (all if empty)'), 0, \State::getType());
             echo "</td>";
             echo "<td width='20%'>";
-            echo "<span id='blocaction2' style='display:none'>";
+            $condition = [];
+            foreach ($CFG_GLPI['inventory_types'] as $inv_type) {
+                $condition['is_visible_' . strtolower($inv_type)] = 1;
+            }
             State::dropdown(
```

Comment:
IMHO, this dropdown should display an `All` option (using the `toadd` param) and when it is selected, the value saved in db should be `["all"]`. This should also be the default value, instead of `[0]`.

Without this `All` option, a user will not able to restore the default behaviour, and, also, a user that opens the form to change another value and save it without changing the current dropdown selection will have the `[]` value stored and it will probably disable the feature.

Reply:
fixed

Reply:
Somewhere in the process, the value is probably an escaped JSON string and could not be decoded.

```php
json_decode('["all"]'); // -> ["all"] (an array)
json_decode('[\"all\"]'); // -> "[\"all\"]" (a string)
```

Reply:
Thanks for the lead I've found

see: [72a869e](https://github.com/glpi-project/glpi/pull/16512/commits/72a869ecba45443e7a702ad7f9d0c0a69c4427ae)